v2/context/context.go

```diff
@@ -50,3 +52,24 @@ func TopicFrom(ctx context.Context) string {
 	}
 	return ""
 }
+
+// Opaque key type used to store backoff parameters
+type retryKeyType struct{}
+
+var retryKey = retryKeyType{}
+
+// WithRetry returns back a new context with the retry backoff parameters.
+func WithRetry(ctx context.Context, backoff types.Backoff) context.Context {
```

Comment:
We might want more than one strategy for backoff, this is linear and we might also want exponential backoff.

Also, rather than a custom type (or an internal custom type is ok) I would not expose that to the user. I recommend being explicit about the backoff and retries:

```
func WithLinearBackoff(ctx context.Context, period time.Duration, maxTries uint) context.Context {
```

is one way. I think internally we should make a type, it can be defined here in this file:

```go
type BackoffConfig struct {
    Strategy string
    Period time.Duration
    MaxRetries int
}
```

And WithLinearBackoff makes a `BackoffConfig{Strategy:"linear", ...`. 

I suggest this to let the api self documenting and not require the caller to understand another object to make.

Reply:
sound good. Fixing....